x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/math/Truncate.java

```diff
@@ -6,36 +6,42 @@
 package org.elasticsearch.xpack.sql.expression.function.scalar.math;
 
 import org.elasticsearch.xpack.sql.expression.Expression;
-import org.elasticsearch.xpack.sql.expression.Literal;
-import org.elasticsearch.xpack.sql.expression.function.scalar.math.BinaryMathProcessor.BinaryMathOperation;
-import org.elasticsearch.xpack.sql.tree.Source;
+import org.elasticsearch.xpack.sql.expression.function.scalar.math.BinaryOptionalMathProcessor.BinaryOptionalMathOperation;
 import org.elasticsearch.xpack.sql.tree.NodeInfo;
-import org.elasticsearch.xpack.sql.type.DataType;
+import org.elasticsearch.xpack.sql.tree.Source;
+
+import java.util.List;
 
 /**
  * Function that takes two parameters: one is the field/value itself, the other is a non-floating point numeric
  * which indicates how the truncation should behave. If positive, it will truncate the number till that
  * parameter count digits after the decimal point. If negative, it will truncate the number till that parameter
  * count digits before the decimal point, starting at the decimal point.
  */
-public class Truncate extends BinaryNumericFunction {
+public class Truncate extends BinaryOptionalNumericFunction {
     
     public Truncate(Source source, Expression left, Expression right) {
-        super(source, left, right == null ? Literal.of(left.source(), 0) : right, BinaryMathOperation.TRUNCATE);
+        super(source, left, right);
     }
 
     @Override
-    protected NodeInfo<Truncate> info() {
+    protected NodeInfo<? extends Expression> info() {
         return NodeInfo.create(this, Truncate::new, left(), right());
     }
 
     @Override
-    protected Truncate replaceChildren(Expression newLeft, Expression newRight) {
-        return new Truncate(source(), newLeft, newRight);
+    protected BinaryOptionalMathOperation operation() {
+        return BinaryOptionalMathOperation.TRUNCATE;
     }
-
+    
     @Override
-    public DataType dataType() {
-        return left().dataType();
+    public Expression replaceChildren(List<Expression> newChildren) {
+        if (right() != null && newChildren.size() != 2) {
```

Comment:
I'd extract this to the parent method and leave here only the instance creation, as it's duplicated also in `Round`.